Automattic/newspack-blocks
# Homepage Posts: Fix issues with translation

### All Submissions:


### Changes proposed in this Pull Request:

This PR fixes a handful of issues with translations in the Homepage Posts block:

* It restores the front-end translation for the block (for the words 'by' and 'and' in the byline). I think this was lost in the recent restructuring.
* It adds a translation for the strings 'by' and 'and' in Spanish to the editor (which were missing previously).
* It updates the date in the editor to display in the correct language. 
* It updates the POT file for the whole block.

Please note: I think the current approach to the translations for the editor are not the correct approach, but I haven't managed to get that to work yet.

The editor should either load a json file of translations with the naming convention `{slug}-{language}-{script handle}.json` (which currently works -- in the plugin, it's `newspack-blocks-es_ES-newspack-blocks-editor.json`), or one with an md5 hash, `{slug}-{language}-{md5}.json` (like `newspack-blocks-es_ES-f03802d9ac1ae8f3b85caee4347b93a7.json`). That hash is dynamically generated as part of the .json file name when you run `wp i18n make-json [translation file]`; one .json file is generated per JavaScript file, and the md5 hash is created from the JavaScript file's original name.

I haven't managed to get `wp_set_script_translations()` to pick up the `{slug}-{language}-{md5}.json` files, though it does work fine for `{slug}-{language}-{script handle}.json`. I'm not sure if it's something to do with how the blocks are structured, or I'm doing something wrong. But to get the translation working in the editor for both the donate and homepage posts blocks, I've copy-pasted what's generated in their individual `{slug}-{language}-{md5}.json` files into `newspack-blocks-es_ES-newspack-blocks-editor.json`. 

This works for now, but definitely won't scale as we add more translations and languages. We might be able to get around it by creating a script to create and combine the files, but I'm really curious why it's not working.

More info here: https://developer.wordpress.org/block-editor/developers/internationalization/

### How to test the changes in this Pull Request:

1. Set up a Homepage Posts block, and a Donate block on a page.
2. Navigate to Settings > General, and switch your site language to Spanish. 
3. View the blocks on the front end; note that the Donate block is translated, but the Homepage Posts block is not (specifically, the author byline):

![image](https://user-images.githubusercontent.com/177561/71206076-99bd2e00-2258-11ea-8023-4bb4978bd162.png)

4. View in the editor, and note a similar issue (except here, the date on the posts block is also not translated):

![image](https://user-images.githubusercontent.com/177561/71206120-b0fc1b80-2258-11ea-9133-e296abb0eb61.png)

5. Apply the PR and run `npm run build`.
6. View the page on the front-end again; confirm that both the Donate block and Homepage Posts block are now translated (the Donate block helps confirm there are no regressions):

![image](https://user-images.githubusercontent.com/177561/71206184-ca9d6300-2258-11ea-9a14-b8d3086c534c.png)

7. View the page in the editor, and confirm that the Donate block is still translated, and that the string 'by' and the date are also both in Spanish:

![image](https://user-images.githubusercontent.com/177561/71206228-e0128d00-2258-11ea-9a9b-fbd438eaf9c8.png)

### Other information:
